text/0000-no-more-extern-crate.md

```diff
@@ -0,0 +1,136 @@
+}
+```
+
+# Reference-Level Explanation
+[reference]: #reference
+
+External crates can be passed to the rust compiler using the
+`--extern CRATE_NAME=PATH` flag.
+For example, `cargo build`-ing a crate `my_crate` with a dependency on `rand`
+results in a call to rustc that looks something like
+`rustc --crate-name mycrate src/main.rs --extern rand=/path/to/librand.rlib ...`.
+
+When an external crate is specified this way,
+the crate will automatically brought into scope as if an
+`extern crate name_of_crate;`
+declaration had been added to the current crate root.
+This behavior won't occur when including a library using the `-l`
+or `-L` flags.
+
+We will continue to support the current `extern crate` syntax,
+both for backwards compatibility and to enable users who want to use manual
+`extern crate` in order to have more fine grained control-- say, if they wanted
+to import an external crate only inside an inner module.
+No automatic import will occur if an `extern crate` declaration for the same
+external dependency appears anywhere within the crate.
+For example, if `rand = "0.3"` is listed as a dependency in Cargo.toml
+and `extern crate rand;` appears somewhere in the crate being compiled,
+then no implicit `extern crate rand;` will be added.
+If Cargo.toml were to also list another dependency, `log = "0.3"`, and no
+`extern crate log;` appears in the crate being compiled,
+then an `extern crate log;` would be implicitly added.
+
+Additionally, items such as modules, types, or functions that conflict with
+the names of implicitly imported crates will cause the implicit `extern crate`
+declaration to be removed.
+Note that this is different from the current behavior of the
+implicitly-imported `std` module.
+Currently, creating a root-level item named `std` results in a name conflict
+error. For consistency with other crates, this error will be removed.
+Creating a root-level item named `std` will prevent `std` from being included,
+and will trigger a warning.
+
+It will still be necessary to use the `extern crate` syntax when using
+`#[macro_use]` to import macros from a crate. This is necessary in order to
+prevent macros from being automatically brought into scope and changing the
+behavior of existing code.
+However, as specified in
+[RFC 1561](https://github.com/rust-lang/rfcs/blob/master/text/1561-macro-naming.md#importing-macros),
+macros 2.0 will no longer require `#[macro_use]`, replacing it with
+normal `use` declarations, for which no `extern crate` is required.
+
+One final remaining use case of `extern crate` syntax is for aliasing, i.e.
+`extern crate foo as bar;`. There is no way to infer aliasing information from
+Cargo.toml, so aliased crates will need to be specied using `extern crate`
+syntax.
+
+# Alternatives
+[alternatives]: #alternatives
+
+- Don't do this.
+- Specify external dependencies using only `extern crate`, rather than only
+`Cargo.toml`, by using `extern crate foo = "0.2";` or similar. This would
```

Comment:
We can do something even simpler. If cargo finds a missing dependency, it can offer to add the latest version from crates.io to `Cargo.toml` (pinning it to `^latestVersion`).